src/main/java/org/apache/maven/plugins/dependency/analyze/AbstractAnalyzeMojo.java

```diff
@@ -357,6 +359,15 @@ private boolean checkDependencies()
             warning = true;
         }
 
+        if ( !testArtifactsWithNonTestScope.isEmpty() )
+        {
+            getLog().warn( "Non-test scoped test only dependencies found:" );
```

Comment:
Is this the message for 'have a broader scope' in the comment above? If so, perhaps change the comment to read more specifically that you are detecting "non-test scoped test-only dependencies'.